src/test/java/duckling/support/MockServerSocket.java

```diff
@@ -0,0 +1,39 @@
+package duckling.support;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketAddress;
+
+public class MockServerSocket extends ServerSocket {
+    private boolean isBound = false;
+
+    public MockServerSocket() throws IOException {
+    }
+
+    @Override
+    public Socket accept() {
+        return new Socket();
+    }
+
+    @Override
+    public void bind(SocketAddress address) {
+        this.isBound = true;
+    }
+
+    @Override
+    public void close() {
+        this.isBound = false;
+    }
+
+    @Override
+    public boolean isBound() {
+        return this.isBound;
+    }
+
+    @Override
+    public boolean isClosed() {
+        return !this.isBound;
+    }
+
+}
```

Comment:
Aha!  Thanks for that hotkey, I'm loving IntelliJ more and more the more I learn.  I bet I can find a setting to hard-wrap my code as well.